pkg/detector/library/maven/compare_test.go

```diff
@@ -64,6 +64,39 @@ func TestComparer_IsVulnerable(t *testing.T) {
 			},
 			want: true,
 		},
+		{
+			name: "version requirements",
+			args: args{
+				currentVersion: "1.2.3",
+				advisory: dbTypes.Advisory{
+					VulnerableVersions: []string{"(,1.2.3]"},
+					PatchedVersions:    []string{">1.2.3"},
```

Comment:
It should be the exact version

```suggestion
					PatchedVersions:    []string{"1.2.4"},
```